### PR TITLE
Kintsugi: Add integration tests.

### PR DESCRIPTION
**Pull request by byohay**
- kintsugi.gemspec: Add git as development dependency.
- Add integration tests.

### CODE REVIEW

File: spec/kintsugi_spec.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    project.new_target("com.apple.product-type.library.static", "bar", :ios)
+    project.save
+    git.add(:all => true)
+    git.commit("Add target bar")
+
+    `git -C #{git_directory_path} rebase #{first_commit_hash} &> /dev/null`
+    Kintsugi::run([File.join(project.path, "project.pbxproj")])
+
+    project = Xcodeproj::Project.open(project_path)
+    expect(project.targets.count).to equal(2)
```
Comment (ashdnazg) [inline spec/kintsugi_spec.rb:53]:
```suggestion
    expect(project.targets.map(&:display_name)).to contain_exactly("foo", "bar")
```

File: spec/kintsugi_spec.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    Kintsugi::run(["driver",
+                   base_project.path + "project.pbxproj",
+                   ours_project.path + "project.pbxproj",
+                   theirs_project.path + "project.pbxproj",
+                   original_project.path + "project.pbxproj"])
+
+    # Changes are supposed to be in `ours_project` after driver is called.
+    ours_project = Xcodeproj::Project.open(ours_project.path)
+    expect(Xcodeproj::Project.open(ours_project.path).targets.count).to equal(2)
```
Comment (ashdnazg) [inline spec/kintsugi_spec.rb:78]:
```suggestion
    expect(ours_project.targets.map(&:display_name)).to contain_exactly("foo", "bar")
```

File: spec/kintsugi_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  end
+
```
Comment (ashdnazg) [inline spec/kintsugi_spec.rb:80]: add tests for errors - make sure the state of the folder in the end is what it should be.
If you can, also add tests for the driver

File: kintsugi.gemspec [hunk 4]
```diff
@@ -28,5 +28,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rspec", "~> 3.9"
   spec.add_development_dependency "rubocop", "1.12.0"
   spec.add_development_dependency "rubocop-rspec", "2.2.0"
+  spec.add_development_dependency "rubocop-performance", "1.13.3"
```
[the remark this answers is not in the document]
Comment (byohay) [reply kintsugi.gemspec:31]: Removed.

File: spec/kintsugi_spec.rb (hunk 3 above)
Comment (byohay) [reply spec/kintsugi_spec.rb:80]: Added a test that checks that after Kintsugi failed to resolve a conflict the project wasn't touched and is still has conflicts.

**Comment by byohay** [comment]
@ashdnazg round done.

File: spec/kintsugi_integration_spec.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      git.add(all: true)
+      git.commit("Change target product name to baz")
+
+      `git -C #{git_directory_path} #{git_command} #{first_commit_hash} &> /dev/null`
+
+      expect {
+        Kintsugi.run([File.join(project.path, "project.pbxproj")])
+      }.to raise_error(Kintsugi::MergeError)
+      expect {
+        project = Xcodeproj::Project.open(project.path)
```
Comment (ashdnazg) [inline spec/kintsugi_integration_spec.rb:125]: is this the best way to check if conflicts are kept?
Maybe just diff the file with the previous version?

**Review by ashdnazg — APPROVED** [review]
One comment.
The rest is really great!

**Pull request merged**